src/solvers/flattening/boolbv_index.cpp

```diff
@@ -16,12 +16,6 @@ Author: Daniel Kroening, kroening@kroening.com
 
 bvt boolbvt::convert_index(const index_exprt &expr)
 {
-  if(expr.id()!=ID_index)
-    throw "expected index expression";
-
-  if(expr.operands().size()!=2)
-    throw "index takes two operands";
-
```

Comment:
Indeed they should but I'd prefer a DATA_INVARIANT for at least the second until we are actually enforcing these properties.